data-raw/generate-comp-data.R

```diff
@@ -0,0 +1,50 @@
+## This script generates the comparison data used in "test-old_new" unit testing.
+## The objective of the test is to do a bit for bit comparison, to make sure that "
+## patch" and "minor" Hector PRs do not impact Hector results. However, when there
+##is an expected "major" Hector development there may be an expected change in Hector behavior.
+##If/when this is case use this script to generate new comparison data.
+
+# Since this script will only be run occasionally while actively developing Hector, use
+# devtools::load_all() instead of loading the built Hector package.
+devtools::load_all()
+
+# Create the output directory.
+out_dir <- here::here('tests', 'testthat', 'compdata')
+dir.create(out_dir, showWarnings = FALSE)
+
+# Determine the version of Hector and the git commit, that generates the comparison data.
+# This will be saved to help keep track of the last time the comparison data
+# was actually updated.
+hector_version <- packageVersion('hector')
+hector_commit  <- system("git rev-parse --short HEAD", intern = TRUE)
+
+
+vars_to_save  <- c(ATMOSPHERIC_CO2(), GLOBAL_TEMP(), RF_TOTAL(), RF_CO2(), HEAT_FLUX())
+dates_to_save <- 1750:2300
```

Comment:
These year values are hard coded.  If these bounds ever change it will be a pain to sort out.  Consider making these variables where one change can be applied.

Reply:
I've updated the `run to dates` to reflect the dates in the `dates_to_save` so that if changes are made to `vars_to_save` and `dates_to_save` it will impact the downstream code. Is that what you mean?